conans/client/generators/cmake_common.py

```diff
@@ -674,7 +687,22 @@ class CMakeCommonMacros:
             # Leave the defaults MD (MDd) or replace them with MT (MTd) but taking into account the
             # debug, forcing MXd for debug builds. It will generate MSVCRT warnings if the dependencies
             # are installed with "conan install" and the wrong build type.
-            if(CONAN_LINK_RUNTIME MATCHES "MT")
+            cmake_policy(GET CMP0091 policy_0091)
+            if(policy_0091 STREQUAL "NEW" AND CONAN_LINK_RUNTIME MATCHES "MT")
```

Comment:
Doesn't look right. Put things under the existing ``CONAN_LINK_RUNTIME MATCHES "MT"``, and inside that, change behavior based on the policy. Also, why only if CMAKE_BUILD_TYPE?

Reply:
I was trying to replicate the exact same behavior than the previous one: https://github.com/conan-io/conan/blob/33652d590a4eb32d0d29d8718c0c624010dae426/conans/client/generators/cmake_common.py#L677
where the flags are replaced depending on the build type.

Reply:
The condition ``if(CONAN_LINK_RUNTIME MATCHES "MT")`` is in both places. It could be common.